### ISSUE TITLE
Remove Mongoose buffering layer

### ISSUE THREAD

**Issue by vkarpov15** [Automattic/mongoose#8702]
Shouldn't be necessary if `useUnifiedTopology` is turned on by default. You can still do `Model.find()` before calling `mongoose.connect()`.

Should fix issues:

#3698
#4979

**Comment by AbdelrahmanHafez** [comment]
I am not quite sure what `useUnifiedTopology` achieves, it feels like a mystery to me.

Is that a native solution for buffering on the native mongodb driver?
Can we expect the same behavior of "initialize models, and run commands, and connect later" by `useUnifiedTopology`?

I liked the buffering behavior mainly because I didn't have to wait until the database is connected to initialize the rest of my code.

**Comment by vkarpov15** [comment]
Yes that is essentially the idea. With `useUnifiedTopology`, when you send an operation like `Model.find()`, the MongoDB driver will try to find a server to send the operation to for `serverSelectionTimeoutMS` milliseconds. If it can't for some reason, like if you haven't called `connect()` or the server is down for longer than `serverSelectionTimeoutMS`, then the operation errors out.

**Comment by AbdelrahmanHafez** [comment]
Thank you for the explanation, nice to see that behavior being achieved natively.

**Comment by vkarpov15** [comment]
We took a closer look at this and it looks like we're going to have to keep buffering, at least partially. The problem is that Mongoose buffering allows you to create models and execute commands _before_ calling `connect()`, which is something the MongoDB driver doesn't allow, even with the PR in :point_up: .

However, keeping this behavior isn't terrible, because the behavior we actually want to get rid of is buffering _after_ initial connection. Mongoose buffering can kick in with cases where Mongoose thinks it is disconnected from the database, which is a problem for cases like #9262 where we may want to still allow queries when Mongoose thinks it is disconnected. This is also a problem because now Mongoose needs to maintain its own separate logic for when operations should be buffered, which may not line up exactly with how the MongoDB driver's server selection loop handles "buffering". 

TLDR; we're going to make a much more minor backwards breaking change that prevents Mongoose from buffering after initial connection, but we'll keep pre-connection buffering as is.
